src/Razor/test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/IntegrationTests/CodeGenerationIntegrationTest.cs

```diff
@@ -155,6 +155,27 @@ public class MyModel
             AssertLinePragmas(compiled.CodeDocument, designTime: false);
         }
 
+        [Fact]
+        public void AttributeDirectiveWithViewImports_Runtime()
+        {
+            // Arrange
+            var projectItem = CreateProjectItemFromFile();
+            AddProjectItemFromText(@"
+@using System
+@attribute [Serializable]");
+
+            // Act
+            var compiled = CompileToAssembly(projectItem, designTime: false, throwOnFailure: false);
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(compiled.CodeDocument.GetDocumentIntermediateNode());
+            AssertCSharpDocumentMatchesBaseline(compiled.CodeDocument.GetCSharpDocument());
+            AssertLinePragmas(compiled.CodeDocument, designTime: false);
+
+            var diagnostics = compiled.Compilation.GetDiagnostics().Where(d => d.Severity >= DiagnosticSeverity.Warning);
+            Assert.Equal("Duplicate 'Serializable' attribute", Assert.Single(diagnostics).GetMessage());
```

Comment:
Is this right? Where's the duplicate?

Reply:
I see it's in the original file. Can we use a different attribute so we don't have this wierdness in the test?

Reply:
I did this on purpose so we also test diagnostics. Didn't seem it worth adding a separate integration test for that

Reply:
ok. Not sure I get it, but I'm not sure it's worth arguing over.